### PR TITLE
chore: DLT-2118 fix lint-staged on commit

### PR DESCRIPTION
**Pull request by braddialpad**
# chore: DLT-2118 fix lint-staged on commit

## Obligatory GIF (super important!)

Obligatory GIF

## :hammer_and_wrench: Type Of Change

These types will not increment the version number, but will still deploy to documentation site on release:

- [x] Other (chore)

## :book: Jira Ticket

https://dialpad.atlassian.net/browse/DLT-2118

## :book: Description

- Fixed lint-staged so that it works again.
- Added linting for JSON files.
- Added unit tests to run on commit on only the tests associated with files you changed via `vitest --related`

## :bulb: Context

Follow up PR to https://github.com/dialpad/dialtone/pull/553, get linting on commit working again.

## :pencil: Checklist

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.
- [x] I have added all relevant documentation.
- [x] I have considered the performance impact of my change.

## :link: Sources

https://vitest.dev/guide/cli#vitest-related

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -17,10 +17,6 @@ Thumbs.db
 *_rsa
 *_rsa.pub
 
-# VS Code #
-.vscode
-*.code-workspace
```
Comment (juliodialpad) [inline .gitignore:22]: Why were we ignoring this ones in first place? those are pretty important files that need to be shared

**Review by juliodialpad — COMMENTED** [review]
Do we need to update the rest of `.eslintrc.cjs` files inside the packages?

File: .gitignore (hunk 1 above)
Comment (braddialpad) [reply .gitignore:22]: Yeah I have no idea, lol. the files inside .vscode were already committed so it makes no sense to have this.

**Comment by braddialpad** [comment]
> Do we need to update the rest of `.eslintrc.cjs` files inside the packages?

Not totally sure actually.. let me check.

**Comment by braddialpad** [comment]
> Do we need to update the rest of `.eslintrc.cjs` files inside the packages?

Nope, seems to inherit it fine without doing that.

**Review by juliodialpad — APPROVED** [review]
Thanks!

**Pull request merged**